action/protocol/staking/endorsement_statemanager.go

```diff
@@ -34,6 +34,12 @@ func (esm *EndorsementStateManager) Put(bucketIndex uint64, endorse *Endorsement
 	return err
 }
 
+// Delete deletes the endorsement of a bucket
+func (esm *EndorsementStateManager) Delete(bucketIndex uint64) error {
+	_, err := esm.DelState(protocol.NamespaceOption(_stakingNameSpace), protocol.KeyOption(endorsementKey(bucketIndex)))
+	return err
+}
```

Comment:
so we plan to delete endorsements that are expired?
should consider keep them (with status = `EndorseExpired `). Currently, withdrawn bucket are deleted, which creates certain level of difficulty for debugging since the data are not available. So deleting data might not be a good design